webui/react/src/components/UPlotChart.tsx

```diff
@@ -83,18 +90,11 @@ const UPlotChart: React.FC<Props> = ({ data, options }: Props) => {
     };
   }, [ chart ]);
 
-  /*
-   * Don't plot the chart if there are no X values.
-   */
-  const hasData: boolean = useMemo(() => {
-    return !!(data && data[0].length > 0);
-  }, [ data ]);
-
-  if (!hasData) {
-    return <Message title="No data to plot." type={MessageType.Empty} />;
-  }
+  useImperativeHandle(ref, () => chart, [ chart ]);
```

Comment:
this will be used in `<LearningCurveChart>` to set focused series when highlighting the table.

Reply:
**question:** would passing in the highlighted series index as a prop work as an alternative?

Reply:
I just think it makes more sense to expose the uPlot instance to be able to access `setSeries` and all those similar methods for stuff that will be tweaked on a per-case basis.

Aiming to have a huge component handling all the tweaks we potentially are gonna need on all the kind of charts look a bit aggressive to me.

On the other hand if we ever will build another chart that needs the same logic that should be shared somehow (no code duplication).